python_modules/dagster/dagster/_core/execution/job_backfill.py

```diff
@@ -58,10 +62,25 @@ def execute_job_backfill_iteration(
 
     partition_set = _get_partition_set(workspace_process_context, backfill)
 
+    # refetch in case the backfill status has changed
+    backfill = cast(PartitionBackfill, instance.get_backfill(backfill.backfill_id))
+    if backfill.status == BulkActionStatus.CANCELING:
+        for all_runs_canceled in cancel_backfill_runs_and_cancellation_complete(
```

Comment:
Surprised that python allows `all_runs_canceled` to retain its value after the for loop?

Regardless the code is working so this seems fine

Reply:
we use this pattern a couple other places in the backfill iteration loop, so i think it should be good! 

example https://github.com/dagster-io/dagster/blob/4a0ae0a7d334b28b32bade088b53b29b6f580d0b/python_modules/dagster/dagster/_core/execution/asset_backfill.py#L1051